### PR TITLE
Remove the battle: allow soft wait for troops option

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
Related to #6161

I hope I didn't break anything.

### CODE REVIEW

File: src/fheroes2/battle/battle_arena.cpp [hunk 1]
```diff
@@ -185,24 +179,15 @@ namespace
...
 
-        if ( firstStage ) {
-            units1.SortFastest();
-            units2.SortFastest();
-        }
-        else {
-            std::reverse( units1.begin(), units1.end() );
-            std::reverse( units2.begin(), units2.end() );
-
-            units1.SortSlowest();
```
Comment (ihhub) [inline src/fheroes2/battle/battle_arena.cpp:201]: Should we remove `SortSlowest()` method?

**Review by ihhub — COMMENTED** [review]
Hi @oleg-derevenetz , I left one comment for code cleanup. Could you please take a look?

File: src/fheroes2/battle/battle_arena.cpp (hunk 1 above)
Comment (oleg-derevenetz) [reply src/fheroes2/battle/battle_arena.cpp:201]: Hi @ihhub yes, I removed it. Don't worry, I always do additional cleanup later following fresh SonarCloud warnings, it usually reports functions and methods that have recently become unused :)

Comment (ihhub) [reply src/fheroes2/battle/battle_arena.cpp:201]: Well done!

**Review by ihhub — APPROVED** [review]

**Review by Branikolog — APPROVED** [review]

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you for this change!

**Pull request merged**